### PR TITLE
feat: add parse arguments for openai tool_calls.

### PR DESCRIPTION
**Pull request by orange-crow**
**Features**

add parse arguments for openai tool_calls.

### CODE REVIEW

File: metagpt/provider/openai_api.py [hunk 1]
```diff
@@ -209,7 +234,14 @@ def get_choice_function_arguments(self, rsp: ChatCompletion) -> dict:
...
             # reponse is code
-            return json.loads(message.tool_calls[0].function.arguments, strict=False)
+            try:
+                return json.loads(message.tool_calls[0].function.arguments, strict=False)
+            except json.decoder.JSONDecodeError as e:
+                error_msg = (
+                    f"Got JSONDecodeError for \n{'--'*40} \n{message.tool_calls[0].function.arguments}, {str(e)}"
+                )
+                logger.error(error_msg)
+                return self._parse_arguments(message.tool_calls[0].function.arguments)
```
Comment (geekan) [inline metagpt/provider/openai_api.py:244]: I don't think that's a good approach. We should either reuse the repair-related method, or we should retry until it succeeds

**Review by geekan — CHANGES_REQUESTED** [review]

File: metagpt/provider/openai_api.py (hunk 1 above)
Comment (orange-crow) [reply metagpt/provider/openai_api.py:244]: Retry until it succeeds sounds good.

Comment (orange-crow) [reply metagpt/provider/openai_api.py:244]: I try it like this, but it does not work.
image
image

And,  I found these in openai functions:
image
When parsing the results of function calls, it seems necessary to use regular expressions for matching.

**Comment by garylin2099** [comment]
severely impacts the ability of using openai response, merged first, recorded todo

**Pull request merged**